### PR TITLE
JAMES-3252 DomainList autoDection should be turned off by default

### PR DESCRIPTION
**Pull request by chibenwa**
Also take this opportunity to clean default domains within all configurations

### CODE REVIEW

File: server/data/data-library/src/main/java/org/apache/james/domainlist/lib/DomainListConfiguration.java [hunk 1]
```diff
@@ -93,13 +93,15 @@ public Builder addConfiguredDomains(Domain... domains) {
 
         public DomainListConfiguration build() {
             return new DomainListConfiguration(
-                autoDetectIp.orElse(true),
-                autoDetect.orElse(true),
+                autoDetectIp.orElse(false),
```
Comment (mbaechler) [inline server/data/data-library/src/main/java/org/apache/james/domainlist/lib/DomainListConfiguration.java:96]: oh, a breaking change! (:

Comment (chibenwa) [reply server/data/data-library/src/main/java/org/apache/james/domainlist/lib/DomainListConfiguration.java:96]: This, is intended! Maybe I should mention it in the changelog?

Comment (mbaechler) [reply server/data/data-library/src/main/java/org/apache/james/domainlist/lib/DomainListConfiguration.java:96]: maybe

**Comment by chibenwa** [comment]
Merged

File: CHANGELOG.md [hunk 2]
```diff
@@ -38,6 +38,7 @@ of tasks being currently executed.
 - Differentiation between domain alias and domain mapping. Read upgrade instructions.
 - JAMES-3122 Log4J2 adoption for Spring product. Log file configuration needs to be updated. See upgrade instructions.
 - JAMES-2760 mailqueue.size.metricsEnabled should be false by default
+- JAMES-3252 DomainList autoDection should be turned off by default. Operators relying on implicit values for enabling DomainList autoDection now needs to explicitly configure it.
```
Comment (rouazana) [inline CHANGELOG.md:41]: autoDection? (twice in this line)

**Pull request closed without merging**